### PR TITLE
add some methods

### PR DESCRIPTION
**Pull request by janehyt**
add some methods which related to authorize_security_group and revoke_security_group

### CODE REVIEW

File: lib/fog/aliyun/models/compute/security_group.rb [hunk 1]
```diff
@@ -40,34 +40,59 @@ def destroy
...
-        # end
+        def authorize_seurity_group_sg_rule(group_id,direction="ingress",options={})
+            requires :id
+            if direction=="egress"
+                service.create_security_group_egress_sg_rule(id,group_id,options)
+            else
+                service.create_security_group_sg_rule(id,group_id,options)
+            end
+                    
+            true
```
Comment (plribeiro3000) [inline lib/fog/aliyun/models/compute/security_group.rb]: Hey, could you use 2 spaces instead of tab?

File: lib/fog/aliyun/models/compute/security_group_rules.rb [hunk 2]
```diff
@@ -12,6 +12,9 @@ def get(security_group_id,options={})
           data = Fog::JSON.decode(service.list_security_group_rules(security_group_id, options).body)
           self.security_group_id = data["SecurityGroupId"]
           data = data["Permissions"]["Permission"]
+          for i in data
+            i["SecurityGroupId"] = self.security_group_id
+          end
```
Comment (plribeiro3000) [inline lib/fog/aliyun/models/compute/security_group_rules.rb]: Please use `each` and try to use better names like:

```ruby
permissions = data["Permissions"]["Permission"]
permissions.each do |permission|
  permission["SecurityGroupId"] = self.security_group_id
end
```

File: lib/fog/aliyun/requests/compute/create_security_group_egress_sg_rule.rb [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          unless priority
+            priority = '1'
+          end
+          parameters["Priority"] = priority
+          pathUrl += '&Priority='
+          pathUrl += priority
+
+          signature = sign(@aliyun_accesskey_secret, parameters)
+          pathUrl += '&Signature='
+          pathUrl += signature
```
Comment (plribeiro3000) [inline lib/fog/aliyun/requests/compute/create_security_group_egress_sg_rule.rb]: This url generation over here sounds like you need to work towards a method to do this based on the models.
I can see this is repeating for every route and is not a good thing. Maybe we could work towards the `as` option in the `attribute` method. Then get the result hash from `all_association_and_attributes` and turn it into this url.
What do you think?

**Comment by plribeiro3000** [comment]
@janehyt Great work over here. Just made some small comments.

Regarding the `attribute` thing let me know if i can help you out

File: lib/fog/aliyun/requests/compute/create_security_group_egress_sg_rule.rb (hunk 3 above)
Comment (janehyt) [inline lib/fog/aliyun/requests/compute/create_security_group_egress_sg_rule.rb]: I tried to do it according to your approach, but found out that there're many things to prepare. If I want to use the attribute method to generate the url, maybe I need to reconstruct models in Compute Module, so I decide not to deal with this issue temporarily.

**Comment by janehyt** [comment]
@plribeiro3000 Thank you for your comments! I've update my code.

File: lib/fog/aliyun/requests/compute/create_security_group_egress_sg_rule.rb (hunk 3 above)
Comment (plribeiro3000) [inline lib/fog/aliyun/requests/compute/create_security_group_egress_sg_rule.rb]: Sure. You are right. 
We can do it in another PR. :+1:

**Comment by plribeiro3000** [comment]
LGTM. Could you rebase this?

PS: i think that maybe you might want to configure your git to set your email so it will give the ownership of this work for you. Right now it is using `sx-h9030` and no email.

Thank your for bearing with me on this!

**Comment by dengqinsi** [comment]
@plribeiro3000 thanks for tackling this!

**Comment by plribeiro3000** [comment]
@dengqinsi Sure. No problem.
